runelite-mixins/src/main/java/net/runelite/mixins/GameScalingMixin.java

```diff
@@ -96,13 +105,58 @@ public void setStretchedKeepAspectRatio(boolean state)
 		cachedStretchedDimensions = null;
 	}
 
+	@Inject
+	@Override
+	public void setScalingFactor(int factor)
+	{
+		if (factor > 100)
+		{
+			factor = 100;
+		}
+		else if (factor < 0)
+		{
+			factor = 0;
+		}
+
+		scalingFactor = (100 - factor) / 100D;
+
+		cachedStretchedDimensions = null;
+		cachedRealDimensions = null;
+
+		// Canvas needs to be resized so setMaxCanvasSize is called with the new realDimensions.
+		setResizeCanvasNextFrame(true);
+	}
+
 	@Inject
 	@Override
 	public Dimension getRealDimensions()
 	{
-		if (isStretchedEnabled() && !isResized())
+		if (isScalingEnabled())
```

Comment:
I would flip this entire if statement and replace it with early return of getCanvas().getSize();